rfcs/00xx-select-from-subscription.md

```diff
@@ -0,0 +1,188 @@
+ pk    |   v   |   is_delete   
+ ------+-------+------------
+ 2     | 2     |    f     
+ 1     | 1     |    t 
+ 1     | 11    |    f 
+(3 rows)
+```
+
+- Similar to the requirement for CURSOR query, `retention` must be specified when creating a subscription, that is eligble for batch select.
+- Unlike CUSOR query, batch select won't see historical snapshot data when the subscription is created. It only sees the data after the subscription is created. Technically the following two queries are equivalent:
+```SQL
+SELECT * FROM sub;
+
+# equivalent to FETCH NEXT from cursor created with since begin() till rw_timestamp == now()
+
+DECLARE cur SUBSCRIPTION CURSOR FOR sub SINCE BEGIN();
+FETCH NEXT FROM cur;
+...
+```
+
+Use case: one-shot inspection on the MV/Table changelog for debugging
+
+
+### Streaming SELECT on subscription
+```SQL
+CREATE TABLE t (pk int PRIMARY KEY, v int);
+
+CREATE SUBSCRIPTION sub FROM t;
+
+CREATE MATERIALIZED VIEW t_changelog AS
+SELECT * FROM sub;
+
+CREATE SINK t_changelog_sink
+AS SELECT t.pk as pk, t.v as v, CASE WHEN t.op = 2 or t.op = 4 THEN TRUE ELSE FALSE END as is_delete FROM sub;
+```
+
+- User can create sink/mv on the subscription to get the changelog stream of the MV/Table the subscription is created upon.
+- `retention` is optional for subscription eligble for streaming query.
+  - When `retention` is set, the corresponding log store will be created and the subscription is eligble for cursor query, batch select query, and streaming query.
+  - When `retention` is not set, no log store will be created for the subscription and the subscription is only eligble for streaming query.
+- A streaming projection executor will be created to convert schema of the upstream stream chunk to `(original schema | op | rw_timestamp)` for streaming query on subscription. Implementation-wise, this is essentially the table value function mentioned in https://github.com/risingwavelabs/rfcs/pull/83.
+
+#### Use-case 1: Alerting MV/Sink based on the first event passing a threshold
+(this example is borrowed from https://github.com/risingwavelabs/rfcs/pull/83)
+```SQL
+  CREATE SOURCE events_source(
+    id: int, 
+    msg: varchar, 
+    status: varchar, 
+    event_time timestamptz, 
+    proc_time timestampz AS proctime()
+  );
+  
+  CREATE MATERIALIZED VIEW events as
+  SELECT * FROM events_source 
+  WHERE proc_time > NOW() - INTERVAL '7 days' 
+  ORDER BY event_time;
+  
+  CREATE MATERIALIZED VIEW windowed_result as 
+  SELECT 
+    window_start,
+    count(*) as all_count,
+    count(*) filter(where status = 'error') as error_count
+  FROM TUMBLE (events, event_time, INTERVAL '1 MINUTES')
+  GROUP BY window_start;
+
+  CREATE SUBSCRIPTION windowed_result_sub from windowed_result;
+
+  # op = 1: insert
+  CREATE MATERIALIZED VIEW error_minutes as 
+  SELECT 
+    distinct on window_start,
+    window_start,
+  FROM windowed_result_sub 
+  WHERE error_count > all_count * 0.9 AND op = 1;
```

Comment:
Off-topic: maybe we can introduce a static variable, e.g, `CHANGE_INSERT` with the value 1 to avoid using magic constants everywhere.